lib/model.d.ts

```diff
@@ -2517,8 +2532,8 @@ export abstract class Model {
    *
    * If changed is called without an argument and no keys have changed, it will return `false`.
    */
-  changed(key: keyof this): boolean;
-  changed(key: keyof this, dirty: boolean): void;
+  changed<K extends keyof this>(key: K): boolean;
```

Comment:
Is this still type safe? Wouldn't this allow any string?

Reply:
It is, in fact you are using this in some of your own methods already 

![alt Proof](https://i.imgur.com/sMzeWRO.png "It works")

Reply:
interesting